src/apps/sketch/components/Board/sketchBoard.jsx

```diff
@@ -15,31 +19,62 @@ export default function SketchBoard(props) {
             x.copy(JSON.parse(message.data))
             othersShapes.current.push(x);
         }
+
+        function onKeyup(e) {
+            switch(e.key){
+                case 'd':
+                case 'D':
+                    // 1st condition is to handle the case while switching from pencle to eraser
+                    touchBrush.current = paletteHandler.getColor() === 'eraser' || !touchBrush.current;
+                    paletteHandler.color(paletteHandler.getColor());
+                    break;
+                case 'e':
+                case 'E':
+                    touchBrush.current = paletteHandler.getColor() !== 'eraser' || !touchBrush.current;
+                    paletteHandler.color('eraser');
+                    break;
+                case '+':
+                    paletteHandler.increaseFont();
+                    break;
+                case '-':
+                    paletteHandler.reduceFont();
+                    break;
+                default:
+                    console.log('');
+            } 
+        }
+        window.addEventListener('keyup', onKeyup);
+
+        return () => window.removeEventListener('keyup', onKeyup);
+
     }, []);
 
     const setup = (p, canvasParentRef) => {
         p.createCanvas(1000, 600).parent(canvasParentRef);
         p.noFill();
-        p.background(225);
+        p.background(DEFAULT_BACKGROUND_COLOR);
     }
 
     const draw = p => {
-        if (p.mouseIsPressed === true) {
+        if (p.mouseIsPressed === true || touchBrush.current) {
             currShape.setFont(props.font);
             currShape.setColor(props.color);
             currShape.setStart(p.mouseX, p.mouseY);
             currShape.setEnd(p.pmouseX, p.pmouseY);
-            const copyObj = new Shape();
-            copyObj.copy(currShape.clone());
-            if(brush)
-                brush.send(currShape.export());
-            currShape.drawShape(p);
+            if(!currShape.isStationary()){
```

Comment:
what does it do?

Reply:
what about  the case when user loops the diagram?

Reply:
our free drawing is a bunch of small straight lines so if u r forming a loop then the start point will be in one line and end point will be in another, 
reason for adding this is, now that we added a shortcut to draw by just clicking on d once, but if he is not moving the cursor the same data will be transerferd mulitple times a sec